[TF 2.0 API Docs] tf.image.convert_image_dtype

imransalam opened the pull request:
Added a usage example in convert_image_dtype under image_ops_impl.py. Also added an attribute error raise and listed it in docstring. The issue has been raised in the link https://github.com/tensorflow/tensorflow/issues/29386

imransalam committed:
[TF 2.0 API Docs] tf.image.convert_image_dtype

Added a usage example in convert_image_dtype under image_ops_impl.py. Also added an attribute error raise and listed it in docstring. The issue has been raised in the link https://github.com/tensorflow/tensorflow/issues/29386

alextp commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -1724,8 +1724,22 @@ def convert_image_dtype(image, dtype, saturate=False, name=None):
+    AttributeError: Type object has no attribute is_integer or is_floating
   """
   image = ops.convert_to_tensor(image, name='image')
+  if hasattr(dtype, 'is_floating') or hasattr(dtype, 'is_integer'):
```
Can you rewrite this to not use hasattr?

imransalam replied:
Sure. Any suggestions?

alextp replied:
if not dtype.is_floating and not dtype.is_integer: raise ?

imransalam replied:
This looks nice. But wouldn't it throw an error at the `if` condition when is_floating attribute does not exists in the dtype.
As an example:

```python
dtype = int
if not dt.is_floating and not dt.is_integer:
  raise AttributeError("Type object has no attribute is_integer or is_floating")
```

This will raise the error

```python
AttributeError                            Traceback (most recent call last)
<ipython-input-10-4228559210de> in <module>()
----> 1 if not dt.is_floating and not dt.is_integer:
      2     raise AttributeError("Type object has no attribute is_integer or is_floating")

AttributeError: type object 'int' has no attribute 'is_floating'
```

alextp commented:
`int` is not a valid tf dtype argument. You can use dtypes.as_dtype(dtype)
to make sure you have a valid dtype

On Tue, Jun 4, 2019 at 11:44 AM Imran Salam <notifications@github.com>
wrote:

> *@imransalam* commented on this pull request.
> ------------------------------
>
> In tensorflow/python/ops/image_ops_impl.py
> <https://github.com/tensorflow/tensorflow/pull/29387#discussion_r290437842>
> :
>
> >    """
>    image = ops.convert_to_tensor(image, name='image')
> +  if hasattr(dtype, 'is_floating') or hasattr(dtype, 'is_integer'):
>
> This looks nice. But wouldn't it throw an error at the if condition when
> is_floating attribute does not exists in the dtype.
> As an example:
>
> dtype = intif not dt.is_floating and not dt.is_integer:
>   raise AttributeError("Type object has no attribute is_integer or is_floating")
>
> This will raise the error
>
> AttributeError                            Traceback (most recent call last)<ipython-input-10-4228559210de> in <module>()----> 1 if not dt.is_floating and not dt.is_integer:
>       2     raise AttributeError("Type object has no attribute is_integer or is_floating")
> AttributeError: type object 'int' has no attribute 'is_floating'
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/29387?email_source=notifications&email_token=AAABHRMMDRL73P5PIYAG6VDPY2ZXHA5CNFSM4HS24SKKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2R7YBY#discussion_r290437842>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABHRMBK6ZIFSFW3NWNSN3PY2ZXHANCNFSM4HS24SKA>
> .
>


-- 
 - Alex

imransalam committed:
Forced conversion of dtype to tf dtype

Removed the raise from docstring and code. Added conversion from any dtype provided to tf type dtypes.

imransalam commented:
@alextp Hey I've done the necessary changes.

alextp commented:
@imransalam this just forces the creation of a dtype object; it doesn't validate that it's floating or integer.

imransalam committed:
added attribute error exception 

Raises attribute error when dtype is neither float nor integer type

imransalam commented:
@alextp added the check

alextp commented on tensorflow/python/ops/image_ops_impl.py:
```diff
@@ -1724,8 +1724,23 @@ def convert_image_dtype(image, dtype, saturate=False, name=None):
   """
   image = ops.convert_to_tensor(image, name='image')
+  dtype = tf.dtypes.as_dtype(dtype)
+  if dtype.is_floating or dtype.is_integer:
```
Can you rewrite this as

if not dtype.is_floating and not dtype.is_integer: raise ...?

Much easier to read without if...pass...else

imransalam replied:
Hey. I made the changes.

imransalam committed:
if condition modified

if condition modified to not use the else part

gbaned commented:
@imransalam Could you please check failed build errors? Thanks!

imransalam committed:
fixed a bug in convert_image_dtype

Removed `tf.` from `dtype = tf.dtypes.as_dtype(dtype)`.

gbaned commented:
@imransalam Can you please address Ubuntu Sanity build failures? Thanks!

imransalam commented:
> @imransalam Can you please address Ubuntu Sanity build failures? Thanks!

I looked at the invocation log,
```
2. do_pylint PYTHON2: Python 2 pylint
  FAIL
3. do_pylint PYTHON3: Python 3 pylint
 FAIL
```
These two have failed. Do you have any idea why they might occur ? Thanks

alextp commented:
The linter is failing. If you search in the logs you'll see the specific
warnings.

On Tue, Jun 11, 2019 at 7:31 AM Imran Salam <notifications@github.com>
wrote:

> @imransalam <https://github.com/imransalam> Can you please address Ubuntu
> Sanity build failures? Thanks!
>
> I looked at the invocation log,
>
> 2. do_pylint PYTHON2: Python 2 pylint
>   FAIL
> 3. do_pylint PYTHON3: Python 3 pylint
>  FAIL
>
> These two have failed. Do you have any idea why they might occur ? Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/29387?email_source=notifications&email_token=AAABHRIA2IKEU7ZHIGMJNW3PZ6ZKHA5CNFSM4HS24SKKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXNJLDI#issuecomment-500864397>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABHROHRDT5Q7E6NRMHCMDPZ6ZKHANCNFSM4HS24SKA>
> .
>


-- 
 - Alex

imransalam committed:
smaller line length 

The length of the line was bigger, hence it was throwing linting errors. Divided into two smaller lines.
